homeassistant/components/kostal_plenticore/switch.py

```diff
@@ -0,0 +1,171 @@
+        self.data_id = data_id
+        self._last_run_success: bool | None = None
+        self._name = name
+        self._is_on = is_on
+        self._attr_name = attr_name
+        self.on_value = on_value
+        self.on_label = on_label
+        self.off_value = off_value
+        self.off_label = off_label
+        self._attr_unique_id = attr_unique_id
+
+        self._device_info = device_info
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return (
+            super().available
+            and self.coordinator.data is not None
+            and self.module_id in self.coordinator.data
+            and self.data_id in self.coordinator.data[self.module_id]
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Register this entity on the Update Coordinator."""
+        await super().async_added_to_hass()
+        self.coordinator.start_fetch_data(self.module_id, self.data_id)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Unregister this entity from the Update Coordinator."""
+        self.coordinator.stop_fetch_data(self.module_id, self.data_id)
+        await super().async_will_remove_from_hass()
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Turn device on."""
+        if await self.coordinator.async_write_data(
+            self.module_id, {self.data_id: self.on_value}
+        ):
+            self._last_run_success = True
+            self.coordinator.name = f"{self.platform_name} {self._name} {self.on_label}"
+            await self.coordinator.async_request_refresh()
+        else:
+            self._last_run_success = False
+
+    async def async_turn_off(self, **kwargs) -> None:
+        """Turn device off."""
+        if await self.coordinator.async_write_data(
+            self.module_id, {self.data_id: self.off_value}
+        ):
+            self._last_run_success = True
+            self.coordinator.name = (
+                f"{self.platform_name} {self._name} {self.off_label}"
+            )
+            await self.coordinator.async_request_refresh()
+        else:
+            self._last_run_success = False
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        return self._device_info
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if device is on."""
+        if self.coordinator.data[self.module_id][self.data_id] == self._is_on:
+            self.coordinator.name = f"{self.platform_name} {self._name} {self.on_label}"
+        else:
+            self.coordinator.name = (
+                f"{self.platform_name} {self._name} {self.off_label}"
+            )
+        return bool(self.coordinator.data[self.module_id][self.data_id] == self._is_on)
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the state attributes."""
+        return {"last_run_success": self._last_run_success}
```

Comment:
Why are we exposing this as a state attribute instead of raising an exception in the service handler?

Reply:
Fixed in https://github.com/home-assistant/core/pull/59998 by removing the property